node/include/cocaine/idl/isolate.hpp

```diff
@@ -0,0 +1,134 @@
+/*
+    Copyright (c) 2011-2014 Andrey Sibiryov <me@kobology.ru>
+    Copyright (c) 2011-2014 Other contributors as noted in the AUTHORS file.
+
+    This file is part of Cocaine.
+
+    Cocaine is free software; you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    Cocaine is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef COCAINE_NODE_SERVICE_ISOLATE_IDL_HPP
+#define COCAINE_NODE_SERVICE_ISOLATE_IDL_HPP
+
+#include <cocaine/rpc/protocol.hpp>
+
+namespace cocaine { namespace io {
+
+struct isolate_tag;
+struct isolate_spawned_tag;
+struct isolate_spooled_tag;
+
+struct isolate {
+    struct spool {
+        typedef isolate_tag tag;
+
+        static const char* alias() {
+            return "spool";
+        }
+
+        typedef boost::mpl::list<
+            std::string
+        > argument_type;
+
+        typedef isolate_spooled_tag dispatch_tag;
+
+        typedef option_of<>::tag upstream_type;
+    };
+
+    struct spawn {
```

Comment:
I think it's a deal for separate feature. I'm still not sure that we even need it.